src/source-af-packet.c

```diff
@@ -1211,13 +1413,16 @@ TmEcode ReceiveAFPLoop(ThreadVars *tv, void *data, void *slot)
                 continue;
             }
         } else if (r > 0) {
-            if (ptv->flags & AFP_RING_MODE) {
-                r = AFPReadFromRing(ptv);
-            } else {
-                /* AFPRead will call TmThreadsSlotProcessPkt on read packets */
-                r = AFPRead(ptv);
-            }
+            r = AFPReadFunc(ptv);
             switch (r) {
+                case AFP_READ_OK:
+                    /* Trigger one dump of stats every second */
+                    TimeGet(&current_time);
```

Comment:
Yes, you read it correctly but we are not supposed to be in this case. Bigger block are supposed to be better 32 which should contains more than 32 packets is the default. time() is supposed to be 3 cycles (if http://stackoverflow.com/questions/6498972/faster-equivalent-of-gettimeofday is correct).

Anyway, easiest solution I see if you want to use get packet info is to add a time_t to PacketThreadVars and pass it to the timestamp of the last packet. If we don't do that on last packet then we have 32 useless affectation. Problem is to get the last packet without doing any test for each packet. If we do a test then we are far over the 3 cycles.